### DISCUSSION TITLE
Run cli programs using txiki and r-tui

### DISCUSSION THREAD

**Discussion by ahaoboy** [saghul/txiki.js#612] [Show and tell]
With a simple polyfill, it is now possible to run r-tui with txiki, here is an example https://github.com/ahaoboy/r-tui-txiki

![cmd](https://github.com/user-attachments/assets/59f73224-216b-4585-a94a-d24f0c474095)

**Comment by saghul** [comment]
This is awesome!

**Comment by lukasMega** [comment]
r-tui is an alternative to [ink](https://github.com/vadimdemedes/ink)? How it differs? i assume it's more lightweight and somehow limited ink alternative?

**Reply by ahaoboy** [reply]
> r-tui is an alternative to [ink](https://github.com/vadimdemedes/ink)? How it differs? i assume it's more lightweight and somehow limited ink alternative?

r-tui is a lightweight implementation of ink, removing yoga-wasm so that it can run on js runtimes that do not support wasm, such as txiki, llrt and rqjs

**Comment by KaruroChori** [comment]
That's great, I was just looking for something like that!
